controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -79,6 +77,47 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// IdentityRef is a reference to an identity to be used when reconciling the managed control plane.
 	// If no identity is specified, the default identity for this controller will be used.
 	IdentityRef *infrav1.AWSIdentityReference `json:"identityRef,omitempty"`
+
+	// Network config for the ROSA HCP cluster.
+	// +optional
+	Network *NetworkSpec `json:"network,omitempty"`
+
+	// The instance type to use, for example `r5.xlarge`. Machine type ref; https://aws.amazon.com/ec2/instance-types/
+	// +optional
+	InstanceType string `json:"instanceType,omitempty"`
```

Comment:
Is there a reason we do not want to support all the fields from the `ROSAMachinePool` here?

Reply:
I believe we give the options for the default MP here, otherwise the user the need to define the RosaMP